Feature/devsu 2457 update kbmatches multiple variants per statement

bnguyen-bcgsc opened the pull request:
- Split Kb Matches table to Kb Matched Statements table that stores statements-related columns and new Kb Matches table that stores only variant-related columns
- Create new Kb Match Join table for many-to-many relationship between kb statements and kb variants
- Declare many-to-many association
- Create new migration that performs data manipulations mentioned above
- Create kbMatch middleware and bundle kbMatches and kbMatchedStatements in wrapper routing component
- Update kbMatch routes
- Create kbMatchedStatement routes
- Update create report tool to allow for legacy format and new format of kb matches and kb matched statements
- Update rapid report filtering
- Add exceptions to schema to accommodate for legacy report kb matches format
- Add unit tests for kb matched statements
- Update unit tests for kbMatches, report, variants, expressions, small mutations, copy number, structural variants
- Update mock report data
- Update swagger spec documentation

Co-authored-by: Lara Matiello Pletz <Nithriel@users.noreply.github.com>

bnguyen-bcgsc committed:
- Split Kb Matches table to Kb Matched Statements table that stores statements-related columns and new Kb Matches table that stores only variant-related columns
- Create new Kb Match Join table for many-to-many relationship between kb statements and kb variants
- Declare many-to-many association
- Create new migration that performs data manipulations mentioned above

Co-authored-by: Lara Matiello Pletz <Nithriel@users.noreply.github.com>

bnguyen-bcgsc committed:
- Update migration file with correct table names and enum type conversion

bnguyen-bcgsc committed:
- Create kbMatch middleware and bundle kbMatches and kbMatchedStatements in wrapper routing component
- Update kbMatch routes
- Update table names in kb match join model
- Update unit tests for kb matches

Co-authored-by: Lara Matiello Pletz <Nithriel@users.noreply.github.com>

bnguyen-bcgsc committed:
- Update create report tool to allow for legacy format and new format of kb matches and kb matched statements
- Add public scope view to kb matched statements routes
- Update rapid report filtering
- Update delete endpoint for kb matched statements to delete only association and not record
- Add exceptions to schema to accommodate for legacy report kb matches format
- Add unit tests for new kb matched statements
- Update rapid report unit tests
- Update mock report data
- Fix typos
- Code linting

Co-authored-by: Lara Matiello Pletz <Nithriel@users.noreply.github.com>

bnguyen-bcgsc committed:
- Test unit test without force: true

Co-authored-by: Lara Matiello Pletz <Nithriel@users.noreply.github.com>

bnguyen-bcgsc committed:
- Update unit tests and variant routes

bnguyen-bcgsc committed:
- Code lint
- Update unit tests

bnguyen-bcgsc committed:
- Update report routes for searching reports by kbMatches and kbMatchedStatements
- Update unit tests

bnguyen-bcgsc committed:
- Code linting

bnguyen-bcgsc committed:
- Update swagger spec doc
- Update unit test lint

bnguyen-bcgsc committed:
Merge branch 'develop' into feature/DEVSU-2457-update-kbmatches-multiple-variants-per-statement

elewis2 commented on app/routes/report/expressionVariants.js:
```diff
@@ -86,7 +86,16 @@ router.route('/')
+                model: db.models.kbMatchedStatements,
+                as: 'kbMatchedStatements',
+                attributes:
+                  ['category'],
```
(noting that this and the same change in copyVariants.js will probably need some updates in the client to handle)

elewis2 commented on app/routes/report/kbMatch/index.js:
```diff
@@ -0,0 +1,15 @@
+
+router.use('/', kbMatches);
+router.use('/:kbMatch/kb-matched-statements', kbMatchedStatements);
+
```
this means there's no way to get a list of kbmatchedStatements for a patient directly, correct?

bnguyen-bcgsc committed:
Merge branch 'develop' into feature/DEVSU-2457-update-kbmatches-multiple-variants-per-statement

bnguyen-bcgsc committed:
Merge branch 'develop' into feature/DEVSU-2457-update-kbmatches-multiple-variants-per-statement

bnguyen-bcgsc committed:
- Update kbmatchedstatement model to include variant hooks for kbmatches, rendering variantId to be whole variant object of the appropriate variant model
- Update /kb-matched-statement endpoints to not be dependent on {kbMatch} ident param
- Update swagger doc

Co-authored-by: Lara Matiello Pletz <Nithriel@users.noreply.github.com>

bnguyen-bcgsc committed:
- Code lint

bnguyen-bcgsc committed:
Merge branch 'develop' into feature/DEVSU-2457-update-kbmatches-multiple-variants-per-statement

bnguyen-bcgsc committed:
- Update kbMatchedStatements unit tests

bnguyen-bcgsc committed:
- Update kbMatchedStatements unit tests

bnguyen-bcgsc committed:
- Fix kbMatches null bug for kbMatchedStatements model hook

Co-authored-by: Lara Matiello Pletz <Nithriel@users.noreply.github.com>

bnguyen-bcgsc committed:
- Add new scope for kbMatchedStatements to not include kbMatches for unit testing
- Update kbMatch unit test route for report.test.js
- Update delete endpoint of kbMatchedStatements to delete bindings with kbMatches on cascade and kbMatches with no associated statements

Co-authored-by: Lara Matiello Pletz <Nithriel@users.noreply.github.com>

bnguyen-bcgsc committed:
- Code lint

bnguyen-bcgsc committed:
- Fix bug where afterFind hook results in error handling bug for kbmatch (when ident leads to nonexisting record)
- Exclude variant ID in nested kbMatches property of kbMatchedStatement
- Fix bug where kbMatchedStatement could not be deleted due to public scope excluding id

Co-authored-by: Lara Matiello Pletz <Nithriel@users.noreply.github.com>

bnguyen-bcgsc committed:
- Update genes and gene viewer endpoint with new kb matched statements response data structure
- Update swagger doc

Co-authored-by: Lara Matiello Pletz <Nithriel@users.noreply.github.com>

bnguyen-bcgsc committed:
- Code lint

bnguyen-bcgsc committed:
Merge branch 'develop' into feature/DEVSU-2457-update-kbmatches-multiple-variants-per-statement

bnguyen-bcgsc committed:
Merge branch 'develop' into feature/DEVSU-2457-update-kbmatches-multiple-variants-per-statement
